### PR TITLE
Some fixes to alignment of meta models

### PR DESCRIPTION
**Pull request by robbr48**
Changes to fetch interface function
- Interfaces no longer existing in external model will be removed from meta model
- Connections to no longer existing interfaces will be removed

Changes to alignment
- Submodels without position and/or rotation are now assumed to be at 0,0,0 when aligning
- Fixed fuzzyEqual function so that it handles zero values

### CODE REVIEW

File: OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp [hunk 1]
```diff
@@ -441,14 +441,14 @@ void MetaModelEditor::addInterfacesData(QDomElement interfaces)
         QDomElement interfacePoint;
         // update interface point
         if (existInterfaceData(subModel.attribute("Name"), interfaceDataElement)) {
-          interfacePoint = getInterfacePoint(subModel.attribute("Name"), interfaceDataElement.attribute("Name"));
-          interfacePoint.setAttribute("Name", interfaceDataElement.attribute("Name"));
+          interfacePoint = getInterfacePoint(subModel.attribute("Name"), interfaceDataElement.attribute("name"));
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: I had updated the attribute in TLM code yesterday. So this code is invalid.
See here https://openmodelica.org/hg/TLMPlugin/rev/26bc3294a9c6

File: OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp [hunk 2]
```diff
@@ -434,6 +434,9 @@ void MetaModelEditor::addInterfacesData(QDomElement interfaces)
   QDomNodeList subModelList = mXmlDocument.elementsByTagName("SubModel");
   for (int i = 0 ; i < subModelList.size() ; i++) {
     QDomElement subModel = subModelList.at(i).toElement();
+    while(!subModel.firstChildElement("InterfacePoint").isNull()) { //First remove all existing interfaces!
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: You don't need to do that. That's a very bad idea.
I already updated the code yesterday to handle both cases i.e., update interface point if its changed OR insert interface point if we don't have any.
https://github.com/OpenModelica/OMEdit/commit/fcbacb97a7450803874c41e2555e9cef1729440a#diff-500e549446a13ad33ae79ea93e5ec16aR441

File: OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp (hunk 1 above)
Comment (robbr48) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: Sorry, didn't notice the TLMPlugin had changed. I will revert his.

File: OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp (hunk 2 above)
Comment (robbr48) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: But what if the user removes an interface from the external model, should it not disappear in the meta model when running fetch interfaces?

Comment (adeas31) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: Oh yeah that's correct. Then we need to see which interfaces are removed externally and just remove them. But what if there are connections associated to them? Should we also remove the connections?

Comment (robbr48) [inline OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp]: I think so, that's what I would expect as a user. I can add this functionality and make a new pull request.

**Pull request merged**